server.js

```diff
@@ -986,32 +990,69 @@ app.post('/submit', (req, res) => {
-    var networkID = await db.collection("networks")
-        .count({ _id: { $lt: network._id } });
-
-    network.networkID = networkID;
-    network.training_count = abbreviateNumber(network.training_count, 4)
-    network.training_steps = abbreviateNumber(network.training_steps, 3)
 
-    var avatar_folder = path.join(__dirname, 'static', 'networks');
-    console.log(avatar_folder);
-    if (!await fs.pathExists(avatar_path)) {
-        await fs.mkdirs(avatar_folder);
+    if (!network) {
+        return res.status(404).render("404");
     }
 
-    var avatar_path = path.join(avatar_folder, network.hash + '.png');
+    var networkID = null;
 
-    if (!await fs.pathExists(avatar_path)) {
-        var retricon = require('retricon-without-canvas');
-        await retricon(network.hash, { pixelSize: 60, }).pngStream().pipe(fs.createWriteStream(avatar_path));
+    // If it's one of the best network, then find it's #
+    if (network.game_count > 0 || network.hash == get_best_network_hash()) {
+        networkID = await db.collection("networks")
+            .count({
+                $and: [
+                    { _id: { $lt: network._id } },
+                    { game_count: { $gt: 0 } }
+                ]
+            });
     }
 
-    res.render('networks/profile', { network, http_host : req.protocol + '://' + req.get('host') });
+    network.networkID = networkID;
+
+    await new Promise(async (resolve, reject) => {
+        var avatar_folder = path.join(__dirname, 'static', 'networks');
 
+        if (!await fs.pathExists(avatar_path)) {
+            await fs.mkdirs(avatar_folder);
+        }
+        
+        var avatar_path = path.join(avatar_folder, network.hash + '.png');
+
+        if (!await fs.pathExists(avatar_path)) {
+            var retricon = require('retricon-without-canvas');
+            retricon(network.hash, { pixelSize: 70, imagePadding: 35, bgColor: '#F0F0F0' })
+                .pngStream()
+                .pipe(fs.createWriteStream(avatar_path))
+                .on('finish', () => {
+                    resolve();
+                });
+        } else {
+            resolve();
+        }
+    });
+    var pug_data = {
+        network,
+        http_host: req.protocol + '://' + req.get('host')
+    };
+
+    if(!req.params.tab || req.params.tab == "matches") {
+        pug_data.matches = await db.collection("matches")
+            .aggregate([ 
+                { "$match": { $or : [{ network1 : network.hash }, { network2 : network.hash }] } },
+                { "$lookup": { "localField": "network2", "from": "networks", "foreignField": "hash", "as": "network2" } }, { "$unwind": "$network2" }, 
+                { "$lookup": { "localField": "network1", "from": "networks", "foreignField": "hash", "as": "network1" } }, { "$unwind": "$network1" }, 
+                { "$sort": { _id: -1 } }, 
+                { "$limit": 100 } 
+            ]).toArray();
+        console.log(pug_data.matches);
```

Comment:
Is this a stray debugging line we can remove?

Reply:
Oh yes please remove. My bad.